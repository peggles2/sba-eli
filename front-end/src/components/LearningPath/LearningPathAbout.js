import React from 'react'
import { Button, Grid, Icon, Modal } from 'semantic-ui-react'

import LearningPathAboutPage from './LearningPathAboutPage'

class LearningPathAbout extends React.Component {
  state = {
    page: 1,
    modalOpen: false
  }
  
  nextPage = () => this.setState({ page: this.state.page + 1 })
  
  setPage = (page) => this.setState({ page: page })
  
  handleOpen = () => this.setState({ modalOpen: true, page: 1 })

  handleClose = () => this.setState({ modalOpen: false })

  render() {
    return (
      <Modal
        trigger={<Button onClick={this.handleOpen}>About</Button>}
        open={this.state.modalOpen}
        onClose={this.handleClose}
      >
        <LearningPathAboutPage page={this.state.page} nextPage={this.nextPage} handleClose={this.handleClose} />
        <Modal.Actions>
          <Grid verticalAlign='middle'>
            <Grid.Row>
              <Grid.Column width={4} textAlign="left">
                { this.state.page < 4 ? <Button onClick={this.handleClose}>Skip</Button> : "" }
              </Grid.Column>
              
              <Grid.Column width={8} textAlign="center">
                <Icon name="circle" onClick={() => this.setPage(1)} style={{marginRight: '10px', color: this.state.page === 1 ? 'green' : 'gray'}} />
                <Icon name="circle" onClick={() => this.setPage(2)} style={{marginRight: '10px', color: this.state.page === 2 ? 'green' : 'gray'}} />
                <Icon name="circle" onClick={() => this.setPage(3)} style={{marginRight: '10px', color: this.state.page === 3 ? 'green' : 'gray'}} />
                <Icon name="circle" onClick={() => this.setPage(4)} style={{color: this.state.page === 4 ? 'green' : 'gray'}} />
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Modal.Actions>
      </Modal>
    )
  }
}

export default LearningPathAbout;