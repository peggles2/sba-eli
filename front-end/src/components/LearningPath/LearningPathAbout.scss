.learning-path-about-icon {
  margin-right: 10px;
}

.learning-path-about-icon-unselected {
  color: gray;
}

.learning-path-about-icon-selected {
  color: green;
}