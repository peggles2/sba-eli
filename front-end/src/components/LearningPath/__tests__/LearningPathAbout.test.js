import React from 'react';
import { Button, Icon, Modal } from 'semantic-ui-react';
import LearningPathAbout from '../LearningPathAbout';
import { shallow } from 'enzyme';

describe('LearningPathAbout', () => {
  it('should properly page by icon clicks', () => {
    const wrapper = shallow(<LearningPathAbout />);
    
    expect(wrapper.find(Icon).length).toBe(4);
    
    expect(wrapper.state("page")).toEqual(1);
    wrapper.find(Icon).at(1).simulate('click');
    expect(wrapper.state("page")).toEqual(2);
    wrapper.find(Icon).at(2).simulate('click');
    expect(wrapper.state("page")).toEqual(3);
    wrapper.find(Icon).at(3).simulate('click');
    expect(wrapper.state("page")).toEqual(4);
    wrapper.find(Icon).at(0).simulate('click');
    expect(wrapper.state("page")).toEqual(1);
  });
  
  it('should display a skip button for first 3 pages', () => {
    const wrapper = shallow(<LearningPathAbout />);
    
    expect(wrapper.find(Button).length).toEqual(1);
    wrapper.find(Icon).at(1).simulate('click');
    expect(wrapper.find(Button).length).toEqual(2);
    wrapper.find(Icon).at(2).simulate('click');
    expect(wrapper.find(Button).length).toEqual(2);
    wrapper.find(Icon).at(3).simulate('click');
    expect(wrapper.find(Button).length).toEqual(1);
  });
});
