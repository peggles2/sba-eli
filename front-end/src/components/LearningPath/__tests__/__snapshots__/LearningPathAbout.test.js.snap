// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningPathAbout should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningPathAbout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "centered": true,
      "children": Array [
        <LearningPathAboutPage
          handleClose={[Function]}
          nextPage={[Function]}
          page={1}
        />,
        <ModalActions>
          <Grid
            verticalAlign="middle"
          >
            <GridRow>
              <GridColumn
                textAlign="left"
                width={4}
              >
                <Button
                  as="button"
                  onClick={[Function]}
                  role="button"
                >
                  Skip
                </Button>
              </GridColumn>
              <GridColumn
                textAlign="center"
                width={8}
              >
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "green",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                    }
                  }
                />
              </GridColumn>
            </GridRow>
          </Grid>
        </ModalActions>,
      ],
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "onClose": [Function],
      "open": false,
      "trigger": <Button
        as="button"
        onClick={[Function]}
        role="button"
      >
        About
      </Button>,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClose": [Function],
          "nextPage": [Function],
          "page": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Grid
            verticalAlign="middle"
          >
            <GridRow>
              <GridColumn
                textAlign="left"
                width={4}
              >
                <Button
                  as="button"
                  onClick={[Function]}
                  role="button"
                >
                  Skip
                </Button>
              </GridColumn>
              <GridColumn
                textAlign="center"
                width={8}
              >
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "green",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                    }
                  }
                />
              </GridColumn>
            </GridRow>
          </Grid>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridRow>
              <GridColumn
                textAlign="left"
                width={4}
              >
                <Button
                  as="button"
                  onClick={[Function]}
                  role="button"
                >
                  Skip
                </Button>
              </GridColumn>
              <GridColumn
                textAlign="center"
                width={8}
              >
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "green",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                      "marginRight": "10px",
                    }
                  }
                />
                <Icon
                  as="i"
                  name="circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "gray",
                    }
                  }
                />
              </GridColumn>
            </GridRow>,
            "verticalAlign": "middle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <GridColumn
                  textAlign="left"
                  width={4}
                >
                  <Button
                    as="button"
                    onClick={[Function]}
                    role="button"
                  >
                    Skip
                  </Button>
                </GridColumn>,
                <GridColumn
                  textAlign="center"
                  width={8}
                >
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "green",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                      }
                    }
                  />
                </GridColumn>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Button
                    as="button"
                    onClick={[Function]}
                    role="button"
                  >
                    Skip
                  </Button>,
                  "textAlign": "left",
                  "width": 4,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "children": "Skip",
                    "onClick": [Function],
                    "role": "button",
                  },
                  "ref": null,
                  "rendered": "Skip",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Icon
                      as="i"
                      name="circle"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "green",
                          "marginRight": "10px",
                        }
                      }
                    />,
                    <Icon
                      as="i"
                      name="circle"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "gray",
                          "marginRight": "10px",
                        }
                      }
                    />,
                    <Icon
                      as="i"
                      name="circle"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "gray",
                          "marginRight": "10px",
                        }
                      }
                    />,
                    <Icon
                      as="i"
                      name="circle"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "gray",
                        }
                      }
                    />,
                  ],
                  "textAlign": "center",
                  "width": 8,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "i",
                      "name": "circle",
                      "onClick": [Function],
                      "style": Object {
                        "color": "green",
                        "marginRight": "10px",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "i",
                      "name": "circle",
                      "onClick": [Function],
                      "style": Object {
                        "color": "gray",
                        "marginRight": "10px",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "i",
                      "name": "circle",
                      "onClick": [Function],
                      "style": Object {
                        "color": "gray",
                        "marginRight": "10px",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "i",
                      "name": "circle",
                      "onClick": [Function],
                      "style": Object {
                        "color": "gray",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centered": true,
        "children": Array [
          <LearningPathAboutPage
            handleClose={[Function]}
            nextPage={[Function]}
            page={1}
          />,
          <ModalActions>
            <Grid
              verticalAlign="middle"
            >
              <GridRow>
                <GridColumn
                  textAlign="left"
                  width={4}
                >
                  <Button
                    as="button"
                    onClick={[Function]}
                    role="button"
                  >
                    Skip
                  </Button>
                </GridColumn>
                <GridColumn
                  textAlign="center"
                  width={8}
                >
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "green",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                      }
                    }
                  />
                </GridColumn>
              </GridRow>
            </Grid>
          </ModalActions>,
        ],
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "onClose": [Function],
        "open": false,
        "trigger": <Button
          as="button"
          onClick={[Function]}
          role="button"
        >
          About
        </Button>,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClose": [Function],
            "nextPage": [Function],
            "page": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Grid
              verticalAlign="middle"
            >
              <GridRow>
                <GridColumn
                  textAlign="left"
                  width={4}
                >
                  <Button
                    as="button"
                    onClick={[Function]}
                    role="button"
                  >
                    Skip
                  </Button>
                </GridColumn>
                <GridColumn
                  textAlign="center"
                  width={8}
                >
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "green",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                      }
                    }
                  />
                </GridColumn>
              </GridRow>
            </Grid>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <GridRow>
                <GridColumn
                  textAlign="left"
                  width={4}
                >
                  <Button
                    as="button"
                    onClick={[Function]}
                    role="button"
                  >
                    Skip
                  </Button>
                </GridColumn>
                <GridColumn
                  textAlign="center"
                  width={8}
                >
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "green",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                        "marginRight": "10px",
                      }
                    }
                  />
                  <Icon
                    as="i"
                    name="circle"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "gray",
                      }
                    }
                  />
                </GridColumn>
              </GridRow>,
              "verticalAlign": "middle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <GridColumn
                    textAlign="left"
                    width={4}
                  >
                    <Button
                      as="button"
                      onClick={[Function]}
                      role="button"
                    >
                      Skip
                    </Button>
                  </GridColumn>,
                  <GridColumn
                    textAlign="center"
                    width={8}
                  >
                    <Icon
                      as="i"
                      name="circle"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "green",
                          "marginRight": "10px",
                        }
                      }
                    />
                    <Icon
                      as="i"
                      name="circle"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "gray",
                          "marginRight": "10px",
                        }
                      }
                    />
                    <Icon
                      as="i"
                      name="circle"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "gray",
                          "marginRight": "10px",
                        }
                      }
                    />
                    <Icon
                      as="i"
                      name="circle"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "gray",
                        }
                      }
                    />
                  </GridColumn>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <Button
                      as="button"
                      onClick={[Function]}
                      role="button"
                    >
                      Skip
                    </Button>,
                    "textAlign": "left",
                    "width": 4,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "button",
                      "children": "Skip",
                      "onClick": [Function],
                      "role": "button",
                    },
                    "ref": null,
                    "rendered": "Skip",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <Icon
                        as="i"
                        name="circle"
                        onClick={[Function]}
                        style={
                          Object {
                            "color": "green",
                            "marginRight": "10px",
                          }
                        }
                      />,
                      <Icon
                        as="i"
                        name="circle"
                        onClick={[Function]}
                        style={
                          Object {
                            "color": "gray",
                            "marginRight": "10px",
                          }
                        }
                      />,
                      <Icon
                        as="i"
                        name="circle"
                        onClick={[Function]}
                        style={
                          Object {
                            "color": "gray",
                            "marginRight": "10px",
                          }
                        }
                      />,
                      <Icon
                        as="i"
                        name="circle"
                        onClick={[Function]}
                        style={
                          Object {
                            "color": "gray",
                          }
                        }
                      />,
                    ],
                    "textAlign": "center",
                    "width": 8,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "i",
                        "name": "circle",
                        "onClick": [Function],
                        "style": Object {
                          "color": "green",
                          "marginRight": "10px",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "i",
                        "name": "circle",
                        "onClick": [Function],
                        "style": Object {
                          "color": "gray",
                          "marginRight": "10px",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "i",
                        "name": "circle",
                        "onClick": [Function],
                        "style": Object {
                          "color": "gray",
                          "marginRight": "10px",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "i",
                        "name": "circle",
                        "onClick": [Function],
                        "style": Object {
                          "color": "gray",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
